vignettes/workflows.Rmd

```diff
@@ -1,208 +1,67 @@
+lang_output <- function(x, lang) {
+  writeLines(c(sprintf("```%s", lang), x, "```"))
+}
+r_output <- function(x) lang_output(x, "r")
+yaml_output <- function(x) lang_output(x, "yaml")
+path <- orderly:::prepare_orderly_example("demo")
+## Even simpler
+unlink(file.path(path, "archive"), recursive = TRUE)
+unlink(file.path(path, "draft"), recursive = TRUE)
+unlink(file.path(path, "data"), recursive = TRUE)
+unlink(file.path(path, "data"), recursive = TRUE)
+remove_files <- setdiff(list.files(file.path(path, "src"), full.names = TRUE), 
+                        file.path(path, "src", c("minimal", "global")))
+for (file in remove_files) {
+  unlink(file, recursive = TRUE)
+}
+unlink(file.path(path, "before.R"))
+unlink(file.path(path, "source.R"))
+unlink(file.path(path, "demo.yml"))
+unlink(file.path(path, "README.md"))
+unlink(file.path(path, "src", "README.md"))
+
+dir_tree <- function(path) {
+  withr::with_dir(path, fs::dir_tree("."))
+}
 ```
 
-This is especially useful if `report-b` has many dependencies, or depends on particular versions of a report.
-
-### Some comments on collaboration
-
-There are a number of features about the system here that might not be obvious at first, but that will aid collaboration.
-
-* Because each report exists in its own directory within `src/` we reduce the potential for conflicts.  In a collaborative analysis where multiple files are shared across analyses, it is easy to have conflicts.  `orderly` forces analyses into separate directories, which means most of the time it is trivial to merge even long-lived branches.  This is particularly important with relatively new users of git.
-* Because every report gets a unique identifier, it is straightforward to combine outputs that were run on different machines.  In the example above this is a centralised system (one canonical source, several destinations) but in theory more complicated distributed workflows are possible.
-* On the central server, the archive directory is an "append only" system; nothing is ever modified.  This seems wasteful at first as there can be lots of copies of a report, but disk space is cheaper than researcher time (and you can remove redundancy with `orderly::orderly_deduplicate`)
-
-## Group of researchers, sharing via server
-
-This is the model that we use within the [MRC Centre for Global Infectious Disease Analysis](https://www.imperial.ac.uk/mrc-global-infectious-disease-analysis), and it does require considerably more setup, but also more reliability and flexibility.
-
-As above, we use a single "source tree", kept on GitHub or similar.  However, someone in the group takes responsibility for setting up a copy of [OrderlyWeb](https://github.com/vimc/orderly-web), our open source, self-hosted orderly server.  The OrderlyWeb server software provides a number of things
+Orderly supports running workflows - these are pre-configured lists of reports which are run in sequence as part of an analysis.
```

Comment:
I think a bit more motivation here would be useful - something the science team might want to add? But basically "Suppose you had a sequence of reports that you always want to run..." show the problem of doing this manually perhaps

Reply:
Tidied that up, asked Susy to take a look too